modules/afsocket/afinet-dest.c

```diff
@@ -525,25 +524,23 @@ afinet_dd_spoof_write_message(AFInetDestDriver *self, LogMessage *msg, const Log
     default:
       g_assert_not_reached();
     }
-  if (success)
-    {
-      if (libnet_write(self->lnet_ctx) >= 0)
-        {
-          /* we have finished processing msg */
-          log_msg_ack(msg, path_options, AT_PROCESSED);
-          log_msg_unref(msg);
 
-          g_static_mutex_unlock(&self->lnet_lock);
-          return TRUE;
-        }
-      else
-        {
-          msg_error("Error sending raw frame",
-                    evt_tag_str("error", libnet_geterror(self->lnet_ctx)));
-        }
+  if (!success)
+    goto finish;
+
+  if ((success = libnet_write(self->lnet_ctx)) < 0)
```

Comment:
This is not an equivalent transformation; `libnet_write` returns a positive or negative integer.